<launch>
    <arg name="map_name" default="sparse_obstacles"/>
    <arg name="view_rviz" default="true"/>
    <arg name="scan" default="/robot0/scan"/>
    <arg name="_base_frame" default="/robot0"/>
    <arg name="map" default="/gmapping/map"/>
    <arg name="cmd_vel" default="robot0/cmd_vel"/>
    <arg name="pose_noise" default="false"/>
    <arg name="lidar_noise" default="false"/>
    <arg name="landmark_noise" default="false"/>
    <arg name="victim_noise" default="true"/>


    <!-- Simulator -->
	<include file="$(find stdr_robot)/launch/robot_manager.launch" />
	
	<node type="stdr_server_node" pkg="stdr_server" name="stdr_server" output="screen" args="$(find stdr_resources)/maps/$(arg map_name).yaml"/>

	<node pkg="tf" type="static_transform_publisher" name="world2map" args="0 0 0 0 0 0  world map 100" />
	
	<include file="$(find stdr_gui)/launch/stdr_gui.launch"/>
	
	<node pkg="stdr_robot" type="robot_handler" name="$(anon robot_spawn)" args="add $(find pa4_student)/simulation/robots/robot_0.yaml 1 2 0" />
	
	<!-- Mapping -->
	<node pkg="gmapping" type="slam_gmapping" name="mapping_node" args="scan:=/robot0/scan _base_frame:=/robot0 map:=/gmapping/map" />
	
	<!-- RVIZ -->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find pa4_student)/search_rescue.rviz"/>
	
	<!-- Landmarks -->
	<node pkg="pa4_student" name="landmark_node" type="landmarks.py" args="$(arg landmark_noise)"/>
	
	<!-- Victims -->
	<node pkg="pa4_student" name="victim_node" type="victims.py" args="$(arg victim_noise)"/>
	
	<!-- Noise -->
	<node pkg="pa4_student" name="slam_noise_node" type="slam_noise.py" args="$(arg pose_noise) $(arg lidar_noise)"/>

    <!-- Navigate the whole environment -->
    <node pkg="pa4_student" type="path_follower_3.py" name="path_follower_node" output="screen" />
    <node pkg="pa4_student" type="drive_around.py" name="drive_around_node" output="screen" />

</launch>
